lib/rules/disallow-variable-names.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Disallows a specified set of variable names.
+ *
+ * Type: `Array`
+ *
+ * Values: Array of strings, which should be disallowed as variable names
+ *
+ * #### Example
+ *
+ * ```js
+ * "disallowVariableNames": ['temp', 'foo']
+ * ```
+ *
+ * ##### Valid
+ *
+ * ```js
+ * var good = 1;
+ * object[fine] = 2;
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * var temp = 1;
+ * object[foo] = 2;
+ * ```
+ */
+
+var assert = require('assert');
+
+module.exports = function() {};
+
+module.exports.prototype = {
+
+    configure: function(disallowVariableNames) {
+        assert(
+            Array.isArray(disallowVariableNames),
+            'disallowVariableNames option requires an array'
+        );
+
+        this._disallowVariableNames = disallowVariableNames;
+    },
+
+    getOptionName: function() {
+        return 'disallowVariableNames';
+    },
+
+    check: function(file, errors) {
+        var disallowVariableNames = this._disallowVariableNames;
+        file.iterateNodesByType('Identifier', function(node) {
+            if (disallowVariableNames.indexOf(node.name) !== -1) {
```

Comment:
My take on the uppercase/lowercase issue was, that one should be able to prohibit the use of ```foo``` but allow the use of ```Foo```. If you want to prohibit the use of ```Foo```, too, you can just add it to the array. Of course, this would still allow ```fOo``` and so on. And thus one could say this leads to exponential complexity...
That said, I can be persuaded on this one :-D
Are there any other opinions on this?

On the object lookup: Yes, the object lookup is probably faster. But this feels a little bit premature opimization like. I am the last person to not improve the code performance wise, but then we should do it right and do performance testing as well. Let me do some experiments with this.
But depending on the uppercase/lowercase issue I can of course implement a object lookup.